pandas/core/generic.py

```diff
@@ -2604,12 +2604,15 @@ def to_sql(
         schema : str, optional
             Specify the schema (if database flavor supports this). If None, use
             default schema.
-        if_exists : {'fail', 'replace', 'append'}, default 'fail'
+        if_exists : {'fail', 'replace', 'append', 'upsert_overwrite', 'upsert_keep'},\
```

Comment:
This may add ambiguity to the `if_exists` argument, as it is referring now to two concepts:
- if **table** exists
- if **row** exists

Would this be better as a separate `if_row_exists` parameter?

Reply:
Also, the documentation says: 

> How to behave if the **table** already exists.

So, the documentation should be updated at the least.

However, I think it would be conceptually cleaner to separate table-existence from row collision-checks.

Reply:
I think you're making a good point.  I will look to incorporate this shortly

Reply:
On your original proposal, you mentioned using a `method` parameter. That seems like a reasonable name/approach :-)

Reply:
@brylie I think it is a good point, but I am less sure about the action required.

I have actually sent a full reply here and now realise it's the wrong place - oups
https://github.com/pandas-dev/pandas/issues/14553#issuecomment-567150956

Reply:
Since the "upsert" method checks on the primary key, how about name the parameter `if_key_exists`?

Reply:
To be explicit, the parameters should be something like:

- `if_table_exists`
- `if_row_exists`

Since renaming `if_exists` to `if_table_exists` is backwords incompatible, perhaps that rename could be avoided or the `if_exists` parameter could be marked as deprecated.

What is blocking us from adding a new argument for `if_row_exists`?